File: tech-start-test-api/src/main/java/com/test/techstarttestapi/model/Product.java
package com.test.techstarttestapi.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.data.annotation.Id;

@NoArgsConstructor
@AllArgsConstructor
@Data
@ToString
public class Product {
    @Id
    private String Id;
    private String Description;
    private Integer productCode;
    private DistributorLocation Distributor;
    private Manufacturer Manufacturer;
}

File: tech-start-test-angular/src/app/model/Product.ts
import { deserialize, deserializeAs } from "cerialize";
import { Distributor } from "./Distibutor";
import { Manufacturer } from "./Manufacturer";

export class Product{
    @deserialize id: string;
    @deserialize productCode: number;
    @deserialize description: string;    
    @deserializeAs(Distributor, 'distributor') distributor: Distributor;
    @deserializeAs(Manufacturer, 'manufacturer') manufacturer: Manufacturer;

    constructor(id: string, productCode: number, description : string, manufacturer: Manufacturer, distributor: Distributor) {
        this.id = id;
        this.productCode = productCode;
        this.description = description;
        this.manufacturer = manufacturer;
        this.distributor = distributor;
      }
}